[netatmo] Bridge status out of sync

Two weather stations went offline:
```
2024-03-12 13:43:10.574 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'netatmo:account:home' changed from ONLINE to OFFLINE (CONFIGURATION_ERROR): Configuration incomplete, please grant the binding to Netatmo Connect.
2024-03-12 13:43:10.576 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'netatmo:weather-station:home:indoor' changed from ONLINE to OFFLINE (BRIDGE_OFFLINE)
2024-03-12 13:43:10.577 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'netatmo:weather-station:home:indoor_small_bathroom' changed from ONLINE to OFFLINE (BRIDGE_OFFLINE)
```

Because of service unavailability (HTTP code 503):
```
2024-03-12 13:43:05.216 [DEBUG] [mo.internal.handler.ApiBridgeHandler] - executeUri GET  https://api.netatmo.com/api/getstationsdata?device_id=station46&get_favorites=false 
2024-03-12 13:43:10.199 [ERROR] [oauth2client.internal.OAuthConnector] - grant type refresh_token to URL https://api.netatmo.com/oauth2/token failed with HTTP response code 503
2024-03-12 13:43:10.297 [DEBUG] [mo.internal.handler.ApiBridgeHandler] - Failed to load access token: Exception in oauth communication, grant type refresh_token: Bad http response, http code 503
2024-03-12 13:43:10.482 [INFO ] [etatmo.internal.servlet.GrantServlet] - Registered Netatmo servlet at '/netatmo/connect/xxx'
2024-03-12 13:43:10.548 [DEBUG] [tatmo.internal.handler.DeviceHandler] - bridgeStatusChanged for bridge netatmo:weather-station:home:indoor to OFFLINE (CONFIGURATION_ERROR): @text/conf-error-grant-needed
2024-03-12 13:43:10.552 [DEBUG] [tatmo.internal.handler.DeviceHandler] - bridgeStatusChanged for bridge netatmo:weather-station:home:indoor_small_bathroom to OFFLINE (CONFIGURATION_ERROR): @text/conf-error-grant-needed
2024-03-12 13:43:10.552 [DEBUG] [tatmo.internal.handler.ModuleHandler] - bridgeStatusChanged for thing netatmo:indoor:home:indoor:indoor2 to OFFLINE (BRIDGE_OFFLINE)
2024-03-12 13:43:10.553 [DEBUG] [tatmo.internal.handler.ModuleHandler] - bridgeStatusChanged for thing netatmo:outdoor:home:indoor:outdoor to OFFLINE (BRIDGE_OFFLINE)
2024-03-12 13:43:10.552 [DEBUG] [tatmo.internal.handler.ModuleHandler] - bridgeStatusChanged for thing netatmo:indoor:home:indoor:indoor3 to OFFLINE (BRIDGE_OFFLINE)
2024-03-12 13:43:10.558 [DEBUG] [tatmo.internal.handler.ModuleHandler] - bridgeStatusChanged for thing netatmo:indoor:home:indoor_small_bathroom:indoor_office to OFFLINE (BRIDGE_OFFLINE)
2024-03-12 13:43:10.558 [DEBUG] [tatmo.internal.handler.ModuleHandler] - bridgeStatusChanged for thing netatmo:outdoor:home:indoor_small_bathroom:outdoor_shed to OFFLINE (BRIDGE_OFFLINE)
2024-03-12 13:43:10.559 [DEBUG] [tatmo.internal.handler.ModuleHandler] - bridgeStatusChanged for thing netatmo:indoor:home:indoor_small_bathroom:indoor_alva to OFFLINE (BRIDGE_OFFLINE)
2024-03-12 13:43:10.559 [DEBUG] [tatmo.internal.handler.ModuleHandler] - bridgeStatusChanged for thing netatmo:indoor:home:indoor_small_bathroom:indoor_linus to OFFLINE (BRIDGE_OFFLINE)
2024-03-12 13:43:10.577 [WARN ] [handler.capability.WeatherCapability] - Error retrieving weather data 'station46' : Not authenticated
```

The status for both weather stations is correct, but the description is debatable (service unavailability probably shouldn't result in configuration error).

After that follows the same - either HTTP status 500 or 503. I noticed also:

```
2024-03-12 14:28:13.292 [TRACE] [mo.internal.handler.ApiBridgeHandler] -  -returned: code [503 Service Unavailable] body <html>
<head><title>503 Service Temporarily Unavailable</title></head>
<body>
<center><h1>503 Service Temporarily Unavailable</h1></center>
<hr><center>nginx</center>
</body>
</html>

2024-03-12 14:28:13.299 [WARN ] [handler.capability.WeatherCapability] - Error retrieving weather data 'station46' : Error deserializing error: Service Unavailable
```

Then suddenly:
```
2024-03-12 14:43:13.308 [DEBUG] [mo.internal.handler.ApiBridgeHandler] - executeUri GET  https://api.netatmo.com/api/getstationsdata?device_id=station46&get_favorites=false 
2024-03-12 14:43:13.312 [TRACE] [mo.internal.handler.ApiBridgeHandler] -  -with headers: Accept-Encoding: gzip, User-Agent: Jetty/9.4.52.v20230823, Authorization: Bearer xxx
2024-03-12 14:43:13.623 [TRACE] [mo.internal.handler.ApiBridgeHandler] -  -returned: code [200 OK] body CUT
2024-03-12 14:43:13.640 [DEBUG] [handler.capability.RefreshCapability] - Module refreshed, next one in 602s
2024-03-12 14:43:13.640 [DEBUG] [tatmo.internal.handler.ModuleHandler] - bridgeStatusChanged for thing netatmo:indoor:home:indoor_small_bathroom:indoor_office to ONLINE
2024-03-12 14:43:13.642 [DEBUG] [tatmo.internal.handler.ModuleHandler] - bridgeStatusChanged for thing netatmo:indoor:home:indoor_small_bathroom:indoor_alva to ONLINE
2024-03-12 14:43:13.642 [DEBUG] [tatmo.internal.handler.ModuleHandler] - bridgeStatusChanged for thing netatmo:outdoor:home:indoor_small_bathroom:outdoor_shed to ONLINE
2024-03-12 14:43:13.642 [DEBUG] [tatmo.internal.handler.ModuleHandler] - bridgeStatusChanged for thing netatmo:indoor:home:indoor_small_bathroom:indoor_linus to ONLINE
```

Event log:
```
2024-03-12 14:43:13.650 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'netatmo:weather-station:home:indoor_small_bathroom' changed from OFFLINE (BRIDGE_OFFLINE) to ONLINE
2024-03-12 14:43:13.654 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'netatmo:indoor:home:indoor_small_bathroom:indoor_office' changed from OFFLINE (BRIDGE_OFFLINE) to UNKNOWN
2024-03-12 14:43:13.657 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'netatmo:indoor:home:indoor_small_bathroom:indoor_linus' changed from OFFLINE (BRIDGE_OFFLINE) to UNKNOWN
2024-03-12 14:43:13.659 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'netatmo:outdoor:home:indoor_small_bathroom:outdoor_shed' changed from OFFLINE (BRIDGE_OFFLINE) to UNKNOWN
2024-03-12 14:43:13.668 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'netatmo:indoor:home:indoor_small_bathroom:indoor_alva' changed from OFFLINE (BRIDGE_OFFLINE) to UNKNOWN
```

Shortly later the external modules came online as well:
```
2024-03-12 14:53:16.031 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'netatmo:indoor:home:indoor_small_bathroom:indoor_office' changed from UNKNOWN to ONLINE
2024-03-12 14:53:16.059 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'netatmo:outdoor:home:indoor_small_bathroom:outdoor_shed' changed from UNKNOWN to ONLINE
2024-03-12 14:53:16.080 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'netatmo:indoor:home:indoor_small_bathroom:indoor_alva' changed from UNKNOWN to ONLINE
2024-03-12 14:53:16.135 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'netatmo:indoor:home:indoor_small_bathroom:indoor_linus' changed from UNKNOWN to ONLINE
```

The problem with the last two series of status updates is that one of the weather stations went offline while the API bridge was still offline:

![image](https://github.com/openhab/openhab-addons/assets/19519842/bd96068f-1623-459d-890d-b09b87883141)

This is inconsistent. And also, the other weather station never came online until I later restarted the binding.

## Expected Behavior

When a weather station is online, its parent bridge should also be online.

## Current Behavior

A weather station thing came be online and working without the bridge being online, and while another hierarchy remains offline altogether:
- bridge: offline
  - weather station 1: online
    - module 1: online
    - module 2: online
  - weather station 2: offline (bridge offline)
    - module 1: offline
    - module 2: offline

## Possible Solution

Account bridge must have been marked offline here:

https://github.com/openhab/openhab-addons/blob/51b63f3045e1928c249a2f1c30ec0424ad820258/bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/ApiBridgeHandler.java#L234-L235

and most likely triggered from `executeUri`:

https://github.com/openhab/openhab-addons/blob/51b63f3045e1928c249a2f1c30ec0424ad820258/bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/ApiBridgeHandler.java#L309-L312

In this case, will it ever come back online if the next call succeeds?

https://github.com/openhab/openhab-addons/blob/51b63f3045e1928c249a2f1c30ec0424ad820258/bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/ApiBridgeHandler.java#L342-L344

Perhaps what is missing is:
```java
updateStatus(ThingStatus.ONLINE);
```

As a consequence, maybe the other weather station is kept offline:

https://github.com/openhab/openhab-addons/blob/51b63f3045e1928c249a2f1c30ec0424ad820258/bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/CommonInterface.java#L221-L222

## Steps to Reproduce

## Context

This is part of general stability issues since OH 4.1. See also #16485.

## Your Environment
* Version used: 4.1.1

--- a/bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/ApiBridgeHandler.java
+++ b/bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/ApiBridgeHandler.java
@@ -340,6 +340,7 @@ public class ApiBridgeHandler extends BaseBridgeHandler {
             logger.trace(" -returned: code {} body {}", statusCode, responseBody);
 
             if (statusCode == Code.OK) {
+                updateStatus(ThingStatus.ONLINE);
                 return deserializer.deserialize(clazz, responseBody);
             }
 

